hphp/ppc64-asm/asm-ppc64.cpp

```diff
@@ -16,8 +16,89 @@
 
 #include "hphp/ppc64-asm/asm-ppc64.h"
 #include "hphp/ppc64-asm/decoder-ppc64.h"
+#include "hphp/runtime/base/runtime-option.h"
+#include "hphp/util/trace.h"
+
+TRACE_SET_MOD(asmppc64);
 
 namespace ppc64_asm {
+VMTOC::~VMTOC() {
+  FTRACE(1, "Number of values stored in TOC: {}\n",
+    std::to_string(m_last_elem_pos));
+}
+
+int64_t VMTOC::pushElem(int64_t elem) {
+  if (m_map.find(elem) != m_map.end()) {
+    return m_map[elem];
+  }
+  auto offset = allocTOC(static_cast<int32_t>(elem & 0xffffffff), true);
+  m_map.insert( { elem, offset });
+  allocTOC(static_cast<int32_t>((elem & 0xffffffff00000000) >> 32));
+  m_last_elem_pos += 2;
+  return offset;
+}
+
+int64_t VMTOC::pushElem(int32_t elem) {
+  if (m_map.find(elem) != m_map.end()) {
+    return m_map[elem];
+  }
+  auto offset = allocTOC(elem);
+  m_map.insert( { elem, offset });
+  m_last_elem_pos++;
+  return offset;
+}
+
+VMTOC& VMTOC::getInstance() {
+  static VMTOC instance;
+  return instance;
+}
+
+intptr_t VMTOC::getPtrVector() {
+  always_assert(m_tocvector != nullptr);
+  return reinterpret_cast<intptr_t>(m_tocvector->base() + INT16_MAX + 1);
+}
+
+int64_t VMTOC::getValue(int64_t index, bool qword) {
+  HPHP::Address addr = reinterpret_cast<HPHP::Address>(
+      reinterpret_cast<intptr_t>(index) + getPtrVector());
```

Comment:
If I change  reinterpret_cast to static_cast my compilation starts to fail. Is that really the problem? Suggestions?